sdk/analytics/src/main/java/com/klaviyo/analytics/networking/requests/PushTokenApiRequest.kt

```diff
@@ -86,19 +67,33 @@ internal class PushTokenApiRequest(
                     VENDOR to VENDOR_FCM
                 )
             )
-        ).also {
-            initialBody = it.toString()
-        }
+        )
     }
 
+    // Update body to include Device metadata whenever the body is retrieved (typically during sending) so the latest data is included
+    override val requestBody: String?
+        get() = body?.apply {
+            optJSONObject(DATA)?.optJSONObject(ATTRIBUTES)?.apply {
+                put(
+                    ENABLEMENT_STATUS,
+                    if (DeviceProperties.notificationPermission) NOTIFICATIONS_ENABLED else NOTIFICATIONS_DISABLED
+                )
+                put(
+                    BACKGROUND,
+                    if (DeviceProperties.backgroundData) BG_AVAILABLE else BG_UNAVAILABLE
+                )
+                put(METADATA, JSONObject(DeviceProperties.buildMetaData()))
+            }
+        }?.toString()
+
     override fun equals(other: Any?): Boolean {
         return when (other) {
-            is PushTokenApiRequest -> initialBody == other.initialBody
+            is PushTokenApiRequest -> body.toString() == other.body.toString()
```

Comment:
Sorry if I missed what's the reasoning behind using toString() vs comparing body to other.body?

Reply:
I didn't actually try that, let me see. My assumption was separate object instances won't appear equal

Reply:
Ahh might compare references?

Reply:
one difficult thing is order:
```
JSONObject("{'keyA': 'value', 'keyB': 1}").toString() == JSONObject("{'keyB': 1, 'keyA': 'value'}").toString() // false
```

Reply:
but yeah, without to string, its a reference check:
```
JSONObject("{'keyA': 'value'}") == JSONObject("{'keyA': 'value'}")
// false
JSONObject("{'keyA': 'value'}").toString() == JSONObject("{'keyA': 'value'}").toString()
// true
```

Reply:
Thanks for the check kenny. Yeah I was surprised the order thing didn't cause my test to fail actually. Unless its just coincidentally passing. I had to add a json compare function to the unit tests for that same reason.